test/IntegrationTestCommon/Factories/WebApplicationFactoryBase.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using AspNetCoreRateLimit;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Infrastructure.EntityFramework.Repositories;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Bit.IntegrationTestCommon.Factories
+{
+    public static class FactoryConstants
+    {
+        public const string DefaultDatabaseName = "test_database";
+        public const string WhitelistedIp = "1.1.1.1";
+    }
+
+    public abstract class WebApplicationFactoryBase<T> : WebApplicationFactory<T>
+        where T : class
+    {
+        /// <summary>
+        /// The database name to use for this instance of the factory. By default it will use a shared database name so all instances will connect to the same database during it's lifetime.
+        /// </summary>
+        /// <remarks>
+        /// This will need to be set BEFORE using the <c>Server</c> property
+        /// </remarks>
+        public string DatabaseName { get; set; } = FactoryConstants.DefaultDatabaseName;
+
+        /// <summary>
+        /// Configure the web host to use an EF in memory database
+        /// </summary>
+        protected override void ConfigureWebHost(IWebHostBuilder builder)
+        {
+            builder.ConfigureAppConfiguration(c =>
+            {
+                c.AddInMemoryCollection(new Dictionary<string, string>
+                {
+                    // Manually insert a EF provider so that ConfigureServices will add EF repositories but we will override
+                    // DbContextOptions to use an in memory database
+                    { "globalSettings:databaseProvider", "postgres" },
+                    { "globalSettings:postgreSql:connectionString", "Host=localhost;Username=test;Password=test;Database=test" },
+                });
+            });
+
+            builder.ConfigureTestServices(services =>
+            {
+                var dbContextOptions = services.First(sd => sd.ServiceType == typeof(DbContextOptions<DatabaseContext>));
+                services.Remove(dbContextOptions);
+                services.AddScoped(_ =>
+                {
+                    return new DbContextOptionsBuilder<DatabaseContext>()
+                        .UseInMemoryDatabase(DatabaseName)
+                        .Options;
+                });
+
+                // QUESTION: The normal licensing service should run fine on developer machines but not in CI
```

Comment:
We should add a `TODO` that we decided to include the license in the pipeline in the future

Reply:
Added!